### ISSUE TITLE
Syntax sugar for ppx let, while, etc.

### ISSUE THREAD

**Issue by hcarty** [facebook/reason#813]
Using 1.4.0 from opam:

```
$ echo 'let%lwt foo = bar' | refmt -print re -parse ml -use-stdin true -is-interface-pp false
[%%lwt let foo = bar];

$ echo 'let%lwt foo = a and baz = b in expr' | refmt -print re -parse ml -use-stdin true -is-interface-pp false
[%lwt
  {
    let foo = a
    and baz = b;
    expr
  }
];

$ echo 'while%lwt true do expr done' | refmt -print re -parse ml -use-stdin true -is-interface-pp false
[%lwt
  while true {
    expr
  }
];
```

and similar results for `try%lwt`, `if%lwt`, etc.

In this case the original OCaml syntax feels better (to me!) than the refmt'd Reason syntax, particularly when working in a larger Lwt or other ppx-using codebase.

**Comment by chenglou** [comment]
@jordwalke @SanderSpies what do you think? What there a concern that it'd make ppx a bit too easy?

**Comment by hcarty** [comment]
For OCaml users who take advantage of extension point shorthand it does give a significant barrier to entry with Reason.  I think Lwt (lwt.ppx) and Async (ppx_let is Jane St.'s IIRC) are the most affected.

**Comment by Charlesetc** [comment]
Just to clarify for future people reading this

The `%lwt` here only affects the first let statement:
```
[%lwt
  {
    let x = one;
    let y = two;
    ...
  }
]
```

**Comment by dannywillems** [comment]
Seems not fixed now. Is there anyone who is working on this?

**Comment by chenglou** [comment]
Not currently

**Comment by ryb73** [comment]
Would love to see this implemented (or some built-in syntactic sugar for monads a la ppx_let)

**Comment by let-def** [comment]
Isn't this implemented by #1703?

**Comment by jordwalke** [comment]
Yes!
